### PR TITLE
fix: add support for results stored in doc.matches

### PR DESCRIPTION
**Pull request by ZiniuYu**
In some situations, the result of the search query is a single document d with all matches stored in d.matches.
The original logic only deals with the case when the results are stored splitly in chunks.
This pr fixes the above issues.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+/junit/
+/tests/junit/
+/docs/chapters/proto/docs.md
+
+# IntelliJ IDEA
+*.iml
+.idea
+
+# test with config in resources
+tests/integration/crud/simple/simple_indexer/
```
Comment (numb3r3) [inline .gitignore]: what's the reason to forcely ignore this folder?

**Review by numb3r3 — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (ZiniuYu) [reply .gitignore]: it is removed

File: simpleranker.py [hunk 2]
```diff
@@ -50,6 +50,8 @@ def rank(self, docs: DocumentArray, parameters: Dict, *args, **kwargs):
 
         for doc in docs[traversal_paths]:
             matches_of_chunks = []
+            matches_of_chunks.extend(doc.matches)
```
Comment (numb3r3) [inline simpleranker.py:53]: to deal with the following case:
```
m.parent_id is None
```

**Review by numb3r3 — CHANGES_REQUESTED** [review]

File: simpleranker.py [hunk 3]
```diff
@@ -50,15 +50,21 @@ def rank(self, docs: DocumentArray, parameters: Dict, *args, **kwargs):
...
+                    matches_of_chunks.append(m)
+                else:
+                    matches_no_parent.append(m)
             doc.matches.clear()
             for chunk in doc.chunks:
                 matches_of_chunks.extend(chunk.matches)
             groups = groupby(
                 sorted(matches_of_chunks, key=lambda d: d.parent_id),
                 lambda d: d.parent_id,
             )
```
Comment (numb3r3) [inline simpleranker.py]: what about

```
    groups = groupby(
                sorted(matches_of_chunks, key=lambda d: d.parent_id or d.id),
                lambda d: d.parent_id or d.id,
            )
```

**Review by numb3r3 — CHANGES_REQUESTED** [review]

File: simpleranker.py (hunk 3 above)
Comment (ZiniuYu) [reply simpleranker.py]: Yes this was my first thought but decided to take a more conservative way. Will double-test

Comment (ZiniuYu) [reply simpleranker.py]: you are right this one actually works

File: simpleranker.py [hunk 4]
```diff
@@ -50,11 +50,14 @@ def rank(self, docs: DocumentArray, parameters: Dict, *args, **kwargs):
 
         for doc in docs[traversal_paths]:
             matches_of_chunks = []
+            for m in doc.matches:
+                matches_of_chunks.extend(m)
```
Comment (numb3r3) [inline simpleranker.py]:
```suggestion
            match_of_chunks.extend(doc.matches)
```

File: simpleranker.py (hunk 3 above)
Comment (ZiniuYu) [reply simpleranker.py]: failing CI, needs more work

File: simpleranker.py (hunk 4 above)
Comment (numb3r3) [reply simpleranker.py]: `matches_of_chunks.extend(m)` is illegal, since `m` is not a list

**Review by numb3r3 — CHANGES_REQUESTED** [review]

File: simpleranker.py (hunk 4 above)
Comment (numb3r3) [reply simpleranker.py]: and also add a test cases to cover this case

File: tests/conftest.py [hunk 5]
```diff
@@ -42,7 +42,7 @@ def documents_chunk_chunk():
                     }
                 )
                 match.scores['cosine'] = NamedScore(value=random.random())
-                match.parent_id = j
+                match.parent_id = str(j)
```
Comment (numb3r3) [inline tests/conftest.py:45]: also add cases with `parent_id=None`

**Review by numb3r3 — CHANGES_REQUESTED** [review]

**Review by numb3r3 — APPROVED** [review]
LGTM

**Pull request merged**